pkg/build/builder/docker.go

```diff
@@ -66,7 +68,14 @@ func (d *DockerBuilder) Build() error {
 	if err != nil {
 		return err
 	}
-	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, initialURLCheckTimeout, os.Stdin, d.gitClient)
+	startTime := unversioned.Now()
```

Comment:
well i think this block is more complicated than it should be.  it should just need to be:

getStartTime
runStepLogic
merge(?)
recordStep/Stage

As it stands, it's 7 lines and i think at least 3 of these lines are unnecessary(variable declaration, storing stopTime temporarily, and the logic related to subStepDuration),  and most invocations wouldn't have the merge step (and the closure wouldn't help with the merge step, i don't think)

In any case, I'm still not convinced creating closures is going to result in more readability (or reduced linecounts), since it means creating wrapper functions around the blocks of code we want to measure, blocks that otherwise serve no purpose, in exchange for a slightly more awkward callstack and return path.